nemo/collections/nlp/data/dialogue/dataset/dialogue_bert_dataset.py

```diff
@@ -82,15 +86,150 @@ def __init__(self, dataset_split: str, dialogues_processor: object, tokenizer, c
+
+        # drive through dataset: other_label=0; assistant dataset: other_label='54'
+        file_path_for_stats = cfg.data_dir+"/train_slot_stats.tsv"
+        # print("============== file_path_for_stats ==========")
+        # print("============== file_path_for_stats ==========")
+        # print("============== file_path_for_stats ==========")
+        # print("============== file_path_for_stats ==========")
+        # print("============== file_path_for_stats ==========")
+        # print("============== file_path_for_stats ==========")
+        # print("============== file_path_for_stats ==========")
+        # print("============== file_path_for_stats ==========")
+        print(file_path_for_stats)
+        with open(file_path_for_stats) as f:
+            other_label = int(f.read().strip().split('\n')[0].split()[0])
+
+        bio_train_slots = [transfer(t, other_label) for t in features[5]]
+        
+        
+
+        self.all_bio_slots = bio_train_slots
+
+        def get_bio_mention_labels(slot_list, bio_list):
+            # slot_list [-1  0  0  0  3  0  0  0 -1 -1 -1]
+            # bio_list  [-1  0  0  0  1  0  0  0 -1 -1 -1]
+            # the function will generate mention's entity lables
+            # bio_train_slots [3]+[0 0 ... 0 0]
+
+            bio_train_slots=[]
+            for idx_ in range(len(slot_list)):
+                if bio_list[idx_]==1:
+                    bio_train_slots.append(slot_list[idx_])
+            add_zero=len(slot_list)-len(bio_train_slots)
+            
+            bio_train_slots = bio_train_slots+[-1]*(add_zero)
+            # bio_train_slots = [0]+bio_train_slots
+            # bio_train_slots = bio_train_slots+[0]*(add_zero-1)
+            return bio_train_slots
+
+        bio_mention_labels = [get_bio_mention_labels(t, bio_list) for t, bio_list in zip(features[5], bio_train_slots)]
+
+        self.bio_mention_labels = bio_mention_labels
+
+        # this may be wrong, we need to create new mention_loss_mask, as the next floowing function
+        def get_mention_loss_mask(bio_mention_labels_list):
+            # bio_mention_labels_list [3]+[0 0 ... 0 0]
+            # mention_loss_mask_list  [1]+[0 0 ... 0 0 ]
+            mention_loss_mask_list= [1 if b!=-1 else 0 for b in bio_mention_labels_list]
+            return mention_loss_mask_list
+
+        # def get_mention_loss_mask(slot_list):
+        #     # slot_list              [-1  0  0  0  3  0  0  0 -1 -1 -1]
+        #     # the function will generate mention's mask
+        #     # mention_loss_mask_list [ 0  0  0  0  3  0  0  0  0  0  0]
+        #     # s>0 which is not -1 , and not 0, would be other classes 1, 2, 3, ...
+        #     mention_loss_mask_list= [1 if s>0 else 0 for s in slot_list]
+        #     return mention_loss_mask_list
+
+        #this may be wrong, check out the next line
+        mention_loss_mask = [get_mention_loss_mask(b_list) for b_list in bio_mention_labels]
```

Comment:
There is a cleaner way to write this using torch.tensor methods such as mention_loss_mask = bio_menton_labels[bio_menton_labels > -1].astype(int) (can't rmb the exact syntax but should be close.